alchemist/alchemist-implementationbase/src/main/java/it/unibo/alchemist/model/implementations/positions/Euclidean2DPosition.java

```diff
@@ -66,6 +66,24 @@ public Euclidean2DPosition normal() {
         return new Euclidean2DPosition(-getY(), getX());
     }
 
+    @NotNull
+    @Override
+    public Euclidean2DPosition coerceAtMost(final double maximumMagnitude) {
+        if (getMagnitude() > maximumMagnitude) {
+            return resized(maximumMagnitude);
+        }
+        return this;
+    }
+
+    @NotNull
+    @Override
+    public Euclidean2DPosition coerceAtLeast(final double minimumMagnitude) {
+        if (getMagnitude() < minimumMagnitude) {
+            return resized(minimumMagnitude);
+        }
+        return this;
+    }
```

Comment:
That's cut/paste :)
and what does coercing mean for a position2D? Is it a method of Vector? Where is it defined?

Reply:
It's defined in `Vector`, I would have implemented it right there but there's a type problem with the curiously recurring template pattern: in `Vector<S : Vector<S>>` the return type of that method is `S`, but if I return `this` I get a `Vector<S>` and not a `S`, even though `S : Vector<S>`.

Reply:
Because this is a `Vector<S>`, not an `S`, for the compiler :)
anyway, since at some point this is forced to be `S`, I believe you can safely cast it.

Reply:
Ok, I'll have to suppress the unchecked cast warning as well.
By the way, we have two more pending conversations: one is on the first block of changes, the other is on this one. You'll have to click "Load more" to show them up.